### PR TITLE
task_1 commit

### PR DESCRIPTION
**Pull request by igorlabzun**

### CODE REVIEW

File: src/com/crypto/service/BruteForceService.java [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+    final String encryptedFile;
+
+    public BruteForceService(String encryptedFile) {
+        this.encryptedFile = encryptedFile;
+    }
+
+    public void bruteForceDecrypt() throws Exception {
+        System.out.println(" Розшифровка методом brute-force:");
+
+        for (int key = 0; key < 26; key++) {
```
Comment (Volodymyr95) [inline src/com/crypto/service/BruteForceService.java]: 26 is magic number

File: src/com/crypto/service/BruteForceService.java [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
+
 public class BruteForceService {
+    final String encryptedFile;
+
+    public BruteForceService(String encryptedFile) {
+        this.encryptedFile = encryptedFile;
+    }
+
+    public void bruteForceDecrypt() throws Exception {
+        System.out.println(" Розшифровка методом brute-force:");
```
Comment (Volodymyr95) [inline src/com/crypto/service/BruteForceService.java]: use english

File: src/com/crypto/service/BruteForceService.java [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+import com.crypto.transformer.TextTransformer;
+
 public class BruteForceService {
+    final String encryptedFile;
+
+    public BruteForceService(String encryptedFile) {
+        this.encryptedFile = encryptedFile;
+    }
+
+    public void bruteForceDecrypt() throws Exception {
```
Comment (Volodymyr95) [inline src/com/crypto/service/BruteForceService.java]: why exception on method level?

File: src/com/crypto/service/DecryptService.java [hunk 4]
```diff
@@ -1,4 +1,21 @@
...
+
 public class DecryptService {
+    final String encryptedFile;
+
+    public DecryptService(String encryptedFile) {
+
+        this.encryptedFile = encryptedFile;
+    }
+
+    public void decryptFile() throws IOException {
```
Comment (Volodymyr95) [inline src/com/crypto/service/DecryptService.java]: exception on method level

File: src/com/crypto/service/EncryptService.java [hunk 5]
```diff
@@ -1,4 +1,24 @@
...
+    final String outputFile;
+    final int key;
+
+    public EncryptService(String inputFile, String outputFile, int key) {
+        this.inputFile = inputFile;
+        this.outputFile = outputFile;
+        this.key = key;
+    }
+
+    public void encryptFile() throws IOException {
```
Comment (Volodymyr95) [inline src/com/crypto/service/EncryptService.java]: exception on method level

File: src/com/crypto/transformer/TextTransformer.java [hunk 6]
```diff
@@ -1,4 +1,20 @@
...
+                encryptedText.append((char) ((character - base + key) % 26 + base));
+            } else {
+                encryptedText.append(character);
+            }
+        }
+        return encryptedText.toString();
+    }
+
+    public static String decrypt(String text, int key) {
+        return encrypt(text, 26 - key); //
```
Comment (Volodymyr95) [inline src/com/crypto/transformer/TextTransformer.java]: 26 is magic number

File: finalProject.iml
Comment (Volodymyr95) [inline finalProject.iml:1]: delete from pr

File: src/Main.java [hunk 7]
```diff
@@ -1,5 +1,8 @@
+import com.crypto.Run;
+
 public class Main {
     public static void main(String[] args) {
+        Run.pet(args);
```
Comment (Volodymyr95) [inline src/Main.java]: don't use static

File: src/com/crypto/Run.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
+package com.crypto;
+
+
+import com.crypto.model.Mode;
+import com.crypto.service.BruteForceService;
+import com.crypto.service.DecryptService;
+import com.crypto.service.EncryptService;
+import com.crypto.util.InputValidation;
+
+
```
Comment (Volodymyr95) [inline src/com/crypto/Run.java]: formatting

File: src/com/crypto/Run.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+public class Run  {
+    private static final int COMMAND_ARGUMENTS_POSITION = 0;
+    private static final int FILE_PATH_ARGUMENTS_POSITION = 1;
+    private static final int KEY_ARGUMENTS_POSITION = 2;
+
+    public static void pet(String[] args) {
```
Comment (Volodymyr95) [inline src/com/crypto/Run.java]: what pet mean?

File: src/com/crypto/Run.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.crypto.model.Mode;
+import com.crypto.service.BruteForceService;
+import com.crypto.service.DecryptService;
+import com.crypto.service.EncryptService;
+import com.crypto.util.InputValidation;
+
+
+
+public class Run  {
```
Comment (Volodymyr95) [inline src/com/crypto/Run.java]: Run - is a good name for method, it's a verb, but for class this name is not valid

File: src/com/crypto/Run.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+                new DecryptService(encryptedFile,decryptedOutputFile,key).decryptFile();
+                break;
+            case BRUTE_FORCE:
+                BruteForceService.bruteForceDecrypt();
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + mode);
+        }
+
+
```
Comment (Volodymyr95) [inline src/com/crypto/Run.java]: formatting

File: src/com/crypto/service/BruteForceService.java [hunk 12]
```diff
@@ -1,4 +1,32 @@
...
+        System.out.println(" Розшифровка методом brute-force:");
+        caesarCipher.setEnglishAlphabet("abcdefghijklmnopqrstuvwxyz");
+        String variable = caesarCipher.getEnglishAlphabet();
+
+
+
+        for (int key = 0; key < variable.length() ; key++) {
+            String encryptedText = FileOperations.readFile(encryptedFile);
+            String decryptedText = TextTransformer.decrypt(encryptedText, key);
+            System.out.println("Зсув " + key + ": " + decryptedText);
```
Comment (Volodymyr95) [inline src/com/crypto/service/BruteForceService.java]: english

File: src/Main.java (hunk 7 above)
Comment (igorlabzun) [reply src/Main.java]: Didn't understand the remark

File: src/com/crypto/Run.java (hunk 8 above)
Comment (igorlabzun) [reply src/com/crypto/Run.java]: Didn't understand the remark

Comment (igorlabzun) [reply src/com/crypto/Run.java]: formatting what

File: src/Main.java [hunk 13]
```diff
@@ -1,5 +1,8 @@
+import com.crypto.OperationRun;
+
 public class Main {
     public static void main(String[] args) {
 
+        OperationRun.run(args);
```
Comment (Volodymyr95) [inline src/Main.java:6]: don't use static

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**